### ISSUE TITLE
On switching into VS Code, everything is frozen for about 2 seconds

### ISSUE THREAD

**Issue by jakebailey** [microsoft/vscode#151995]
Issue Type: <b>Performance Issue</b>

- Clone https://github.com/microsoft/TypeScript (what I am working on)
- Open `parserArrowFunctionExpression8.ts`
- Switch to a different window
- Switch back to VS Code

CPU profile:  [CPU-20220613T160952.cpuprofile.txt](https://github.com/microsoft/vscode/files/8895085/CPU-20220613T160952.cpuprofile.txt)

This appears to be the delay: 
![image](https://user-images.githubusercontent.com/5341706/173462200-e830e389-8e81-4937-8d54-101bd67f8c90.png)

VS Code version: Code 1.68.0 (4af164ea3a06f701fe3e89a2bcbb421d2026b68f, 2022-06-08T11:44:16.822Z)
OS version: Windows_NT x64 10.0.22000
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-10900K CPU @ 3.70GHz (20 x 3696)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|63.78GB (50.81GB free)|
|Process Argv|--disable-extensions --crash-reporter-id ffd813dd-a264-4f03-8c00-e2403d747bb6|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    0	   121	  7592	code main
    0	    94	 18764	   issue-reporter
    0	    41	 21920	   utility-network-service
    1	   905	 22216	   window (parserArrowFunctionExpression8.ts - TypeScript - Visual Studio Code)
    1	   170	 22592	   gpu-process
    0	   105	 24492	   extensionHost
    0	   117	 14492	     "C:\Users\jabaile\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node --max-old-space-size=3072 "c:\Users\jabaile\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js" --serverMode partialSemantic --useInferredProjectPerProjectRoot --disableAutomaticTypingAcquisition --cancellationPipeName C:\Users\jabaile\AppData\Local\Temp\vscode-typescript\5955710e51061086f842\tscancellation-0121a2507ebaaa5d2976.tmp* --locale en --noGetErrOnBackgroundUpdate --validateDefaultNpmLocation --useNodeIpc
    0	    81	 17144	     "C:\Users\jabaile\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node "c:\Users\jabaile\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\json-language-features\server\dist\node\jsonServerMain" --node-ipc --clientProcessId=24492
    0	   142	 30304	     "C:\Users\jabaile\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node --max-old-space-size=3072 "c:\Users\jabaile\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js" --useInferredProjectPerProjectRoot --enableTelemetry --cancellationPipeName C:\Users\jabaile\AppData\Local\Temp\vscode-typescript\5955710e51061086f842\tscancellation-6ead2e57633a00851068.tmp* --locale en --noGetErrOnBackgroundUpdate --validateDefaultNpmLocation --useNodeIpc
    0	    94	 30488	       "C:\Users\jabaile\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node "c:/Users/jabaile/AppData/Local/Programs/Microsoft VS Code/resources/app/extensions/node_modules/typescript/lib/typingsInstaller.js" --globalTypingsCacheLocation C:/Users/jabaile/AppData/Local/Microsoft/TypeScript/4.7 --enableTelemetry --typesMapLocation "c:/Users/jabaile/AppData/Local/Programs/Microsoft VS Code/resources/app/extensions/node_modules/typescript/lib/typesMap.json" --validateDefaultNpmLocation
    0	   331	 24584	   window (undefined)
    0	    27	 28516	   crashpad-handler
    0	   119	 30204	   shared-process
    0	    80	  8980	     ptyHost
    0	     7	 29704	       console-window-host (Windows internal process)
    0	   102	 30128	       "C:\Program Files\PowerShell\7\pwsh.exe"
    0	    78	 30652	     fileWatcher
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (parserArrowFunctionExpression8.ts - TypeScript - Visual Studio Code)
|    Folder (TypeScript): more than 20532 files
|      File types: js(10404) ts(1788) map(1683) txt(1525) symbols(686)
|                  types(686) jsx(308) json(28) tsbuildinfo(19) md(15)
|      Conf files: github-actions(14) dockerfile(2) devcontainer.json(1)
|                  launch.json(1) settings.json(1) tasks.json(1) gulp.js(1)
|                  package.json(1)
|      Launch Configs: node(2);
```

</details>
Extensions disabled<details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonvspyt551:30345470
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
vscorecescf:30445987
pythondataviewer:30285071
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30449410
vscscmwlcmt:30465135
cppdebug:30492333
vsclangdf:30486550

```

</details>

**Comment by jrieken** [comment]
The file is very small (2 lines) which makes me believe this isn't outline. @jakebailey do you use outline? can you reproduce after hiding outline?

The profile shows some explorer bits (mergeWithLocal) and otherwise mostly tree logic...

<img width="661" alt="Screenshot 2022-06-15 at 16 16 34" src="https://user-images.githubusercontent.com/1794099/173849834-1f6b239f-0bc1-4337-ba63-13818940b2fd.png">

**Comment by jrieken** [comment]
@jakebailey I cannot reproduce (on mac, with dev-setup)... Do you have any special file filtering rules in place? Or do you use the explorer file nesting feature? Or when not in the VS Code windows do you run some command that produces many file events?

**Comment by lramos15** [comment]
Without switching windows does clicking the little refresh icon at the top of the explorer cause it to freeze too?

**Comment by jakebailey** [comment]
Yes, yes it does.

**Comment by lramos15** [comment]
Do you have any `files.exclude` or `explorer.excludeGitIgnore` setting?

**Comment by jakebailey** [comment]
No, I don't have any of those settings set (in my user settings or in the workspace).

I tried to run VS Code out of this repo to do a bisect, since I only started experiencing this in 1.68,  however I am unable to reproduce in a dev setup. I also cannot reproduce this in another repo, like VS Code's; I suspect this is something to do with just how huge the file list is in the TS repo. I also cannot reproduce this on a freshly installed Insiders.

When I was looking for related issues, I found at least a few people saying that they deleted all of their vscode user data and recreated it, which fixed issues like that, but I didn't really want to try that as I'm sure there's a bunch of stuff I'd have to fix to get back into a productive state.
